Running in Terraform Cloud, backup setting is not detected, enabling backups segfaults

<!-- Please keep this line, it helps us automate tagging issues. [issue-type:bug-report] -->

### Terraform Version

```
Terraform v1.0.4
on linux_amd64
```

Running on Terraform Cloud with  v1.20.0 of the linode provider.

### Effected Resource(s)

- linode_instance

### Terraform Configuration Files

<!--
Please put your Terraform configuration files in terraform code blocks like the one below.

If your configuration is too complex to be put into code blocks, consider using a Github Gist or a service like Dropbox and share a link to a zip file containing your configuration.
-->

```terraform
resource "linode_instance" "dev" {
  label = "dev"
  backups_enabled = true
  region = "us-southeast"
  type = "g6-nanode-1"
  tags = [
    "dev",
  ]

  config {
    kernel       = "linode/direct-disk"
    label        = "Installer"
    root_device  = "/dev/sdc"

    devices {
      sda {
        disk_label = "NixOS"
      }

      sdb {
        disk_label = "512 MB Swap Image"
      }

      sdc {
        disk_label = "Installer"
      }
    }

    helpers {
      devtmpfs_automount = false
      distro             = false
      modules_dep        = false
      network            = false
      updatedb_disabled  = false
    }
  }
  config {
    kernel       = "linode/grub2"
    label        = "NixOS"
    root_device  = "/dev/sda"

    devices {
      sda {
        disk_label = "NixOS"
      }

      sdb {
        disk_label = "512 MB Swap Image"
      }
    }

    helpers {
      devtmpfs_automount = false
      distro             = false
      modules_dep        = false
      network            = false
      updatedb_disabled  = false
    }
  }

  disk {
    filesystem       = "swap"
    label            = "512 MB Swap Image"
    read_only        = false
    size             = 512
  }
  disk {
    filesystem       = "ext4"
    label            = "Installer"
    read_only        = false
    size             = 1024
  }
  disk {
    filesystem       = "ext4"
    label            = "NixOS"
    read_only        = false
    size             = 24064
  }
}
```


### Panic Output
```
Stack trace from the terraform-provider-linode_v1.20.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x175a17f]

goroutine 64 [running]:
github.com/linode/terraform-provider-linode/linode.resourceLinodeInstanceUpdate(0x1db68e8, 0xc00007df80, 0xc0007fc780, 0x181a020, 0xc000610400, 0xc0008736a0, 0xab986a, 0xc0007f87a0)
	github.com/linode/terraform-provider-linode/linode/resource_linode_instance.go:1218 +0xfdf
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000575340, 0x1db6878, 0xc000689b00, 0xc0007fc780, 0x181a020, 0xc000610400, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:364 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000575340, 0x1db6878, 0xc000689b00, 0xc000135dc0, 0xc0007f87a0, 0x181a020, 0xc000610400, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:464 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000222150, 0x1db6878, 0xc000689b00, 0xc00063b5e0, 0xc000689b00, 0x1a25e60, 0xc0007d7300)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc0004206e0, 0x1db6920, 0xc000689b00, 0xc000135c00, 0xc0004206e0, 0xc0007d7380, 0xc000501ba0)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1a25e60, 0xc0004206e0, 0x1db6920, 0xc0007d7380, 0xc000201c20, 0x0, 0x1db6920, 0xc0007d7380, 0xc0007e0000, 0x14a6)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00044c000, 0x1dc9e98, 0xc000466a80, 0xc0006cd200, 0xc000203710, 0x280e960, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00044c000, 0x1dc9e98, 0xc000466a80, 0xc0006cd200, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000040510, 0xc00044c000, 0x1dc9e98, 0xc000466a80, 0xc0006cd200)
	google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-linode_v1.20.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

First, the linode provider should properly detect if backups are enabled or not. Second, the linode provider should enable backups when requested.
### Actual Behavior

The provider fails to detect that backups are enabled. When run with already backups enabled and requesting that backups be enabled, the provider segfaults.


### Steps to Reproduce

1. create a linode instance in the web UI with backups enabled
2. import existing linode instance using `terraform import`
3. ensure `backups_enabled = true` in the tf resource definition
4. apply the new plan (tf will report backups as disabled during planning and crash during apply)

In my case, my existing image was a NixOS image I set up.

--- a/linode/resource_linode_instance.go
+++ b/linode/resource_linode_instance.go
@@ -770,6 +770,8 @@ func resourceLinodeInstanceRead(ctx context.Context, d *schema.ResourceData, met
 	flatBackups := flattenInstanceBackups(*instance)
 
 	d.Set("backups", flatBackups)
+	d.Set("backups_enabled", instance.Backups.Enabled)
+
 	d.Set("specs", flatSpecs)
 	d.Set("alerts", flatAlerts)
 

